task2.c

```diff
@@ -0,0 +1,84 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <locale.h>
+
+int main()
+{
+    setlocale(LC_ALL, "ru");
+    printf("Эта программа ищет Седловые точки матрицы.\n\n");
+    FILE* f;
+    if (fopen_s(&f, "matrix.txt", "r") != 0)// открываем файл для чтения
+    {
+        printf("Ошибка открытия файла!\n");
+        return -1;
+    }
+    int rows, cols = 0;
+    if (fscanf_s(f, "%d %d\n", &rows, &cols) < 2 || rows < 1 || cols < 1)  // читаем кол-во строк и столбцов
+    {
+        printf("Кол-во строк или столбцов не может быть меньше 1 (или их нет)!\n");
+        return -1;
+    }
+    int* matrix = (int*)
+        malloc(rows * cols * sizeof(int));  // Читаем данные из файла
+
+    for (int j = 0; j < rows; ++j)
+    {
+        for (int i = 0; i < cols; ++i)
+        {
+            if (fscanf_s(f, "%d", &matrix[i + j * cols]) < 1)
+            {
+                printf("Ошибка чтения (мало данных?)!\n");
+                return -1;
+            }
+        }
+    }
+    for (int j = 0; j < rows; ++j)
+    {
+        int min, max;  // мин\макс значения в строке
+        min = matrix[j * cols];
+        max = min;

```

Comment:
```suggestion
        int min = matrix[j * cols];
        int max = min;
```
Всегда инициализируй переменные при объявлении. Если эти два действия разделять, то в процессе разработки можно выстрелить себе в ногу - написать что-то содержательное между объявлением и инициализацией.